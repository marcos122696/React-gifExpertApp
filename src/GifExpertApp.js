import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () =>{

    const [categories, setCategories] = useState(['Los Simpsons']);

    // const handleAdd = () => {
    //     setCategories([...categories, 'X-MEN']);
    // }


    return (
        <div className="container">
            <div className="header">
                <h2 className="animate__animated animate__rotateInDownLeft">GifMarcosApp</h2>
                <p>By Marcos Gomez</p>
            </div>
            <AddCategory setCategories= { setCategories }/>
            <hr />      
            <ol>
                {
                    categories.map( category => ( // con los () ya esta retornando sin poner el "return"
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>
        </div>
    );  
};