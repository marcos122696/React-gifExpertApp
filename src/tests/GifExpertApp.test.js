import React from 'react';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';


describe('Pruebas en <GifExpertApp.js />', () => {

    const wrapper = shallow(<GifExpertApp  />);

    test('Debe de mostrar el componente <GifExpertApp /> correctamente', () => { 
        expect(wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {

        const categories = ['los simpons', 'dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
        expect(wrapper ).toMatchSnapshot();
        expect( wrapper .find('GifGrid').length ).toBe( categories.length );
    });


});