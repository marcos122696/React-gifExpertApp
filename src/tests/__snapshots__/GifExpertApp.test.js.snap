// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp.js /> Debe de mostrar el componente <GifExpertApp /> correctamente 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <h2
      className="animate__animated animate__rotateInDownLeft"
    >
      GifExpertApp
    </h2>
    <p>
      By Marcos Gomez
    </p>
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas en <GifExpertApp.js /> Debe de mostrar una lista de categorias 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <h2
      className="animate__animated animate__rotateInDownLeft"
    >
      GifExpertApp
    </h2>
    <p>
      By Marcos Gomez
    </p>
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="los simpons"
      key="los simpons"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</div>
`;
